packages/babel-parser/src/parser/location.js

```diff
@@ -125,6 +125,7 @@ export const Errors = Object.freeze({
     "Topic reference was used in a lexical context without topic binding",
   PrimaryTopicRequiresSmartPipeline:
     "Primary Topic Reference found but pipelineOperator not passed 'smart' for 'proposal' option.",
+  PrivateInExpectedIn: "Expected `in` keyword to follow private name",
```

Comment:
Maybe we could also suggest how it should be used?
```suggestion
  PrivateInExpectedIn: "Private names are only allowed in property accesses (`obj.#%0`) or in `in` expressions (`#%0 in obj`)",
```